src/renderer/mod.rs

```diff
@@ -686,3 +728,54 @@ impl<'a> Renderer<'a> {
         Ok(output)
     }
 }
+
+/// Return a Vec of all substrings contained in '[ ]'s
+/// Ignore quoted strings and integers.
+fn pull_out_square_bracket(s: &str) -> Vec<String> {
+    let mut chars = s.chars();
+    let mut results = vec![];
+    loop {
+        match chars.next() {
+            Some('[') => {
+                let c = chars.next().unwrap();
+                if c != '"' && c != '\'' {
+                    let mut inside_bracket = vec![c];
+                    let mut bracket_count = 1;
+                    loop {
+                        let c = chars.next();
+                        match c {
+                            Some(']') => bracket_count -= 1,
+                            Some('[') => bracket_count += 1,
+                            Some(_) => (),
+                            None => break,
+                        };
+                        if bracket_count == 0 {
+                            // Only store results which aren't numbers
+                            let sub :String = inside_bracket.into_iter().collect();
+                            if sub.parse::<usize>().is_err() {
+                                results.push(sub);
+                            }
+                            break;
+                        }
+                        inside_bracket.push(c.unwrap());
+                    }
+                }
+            },
+            None => break,
+            _ => (),
+        }
+    }
+    results
+}
+
+#[test]
+fn test_pull_out_square_bracket() {
```

Comment:
@Keats should this test be moved with the ones below as well? - I'm conscious there were no other tests in this file

Reply:
I think the tests and the function itself can be moved to another file in the module (`square_brackets.rs` or something similar). This test is only testing the function directly so it's ok to leave it next to the function

Reply:
moving to new file